### DISCUSSION TITLE
cosmos-sdk based rollup integration with wallets

### DISCUSSION THREAD

**Discussion by tzdybal** [rollkit/rollkit#130] [General]
We need to think about integrating cosmos-sdk based, optimint rollups with wallets like Kepr.

What are the implications for UI (few stages of confirations, etc)?
What to do about rollbacks?

**Comment by adlerjohn** [comment]
Rollbacks should never happen if the full node the wallet connects to is actually a full node (which it should be). If they're using an Infura-like service (which they should be), then the wallet can just assume the service will never lie, and if they do and you have to do a rollback just nuke the entire wallet state and re-initialize it.

**Comment by adlerjohn** [comment]
"Confirmations" can simply be displayed immediately once accepted by the block producer of the respective block. This might require additional networking messages so that the block producer can communicate to service providers that it will accept a transaction into a block. Note that this doesn't cause any rollbacks of block height because they're just soft confirmations displayed to the user as having their transaction accepted.
